File: 0524-longest-word-in-dictionary-through-deleting/0524-longest-word-in-dictionary-through-deleting.py
class Solution:
    def findLongestWord(self, s: str, dictionary: List[str]) -> str:
        def check(s1, s2):
            j = 0
            for i in range(len(s1)):
                if s1[i] == s2[j]:
                    j+=1
                    if j == len(s2):
                        return True
            return False
        
        a = -1
        for i,v in enumerate(dictionary):
            if check(s, v):
                if a == -1:
                    a = i
                else:
                    if len(dictionary[a]) < len(v):
                        a = i
                    elif len(dictionary[a]) == len(v):
                        if dictionary[a] > v:
                            a =i
        if a == -1:
            return ""
        return dictionary[a]

File: 0524-longest-word-in-dictionary-through-deleting/0524-longest-word-in-dictionary-through-deleting.java
class Solution {
    public String findLongestWord(String s, List<String> dictionary) {
        String result = "";

        for (String d : dictionary) {
            int firstPointer = 0;
            for (int i = 0; i < s.length(); i++ ) {
                if (s.charAt(i) == d.charAt(firstPointer)) {
                    firstPointer++;
                    if (firstPointer == d.length()) {
                        if (result.length() < d.length()) {
                            result = d;
                        } else if ((result.length() == d.length()) && (result.compareTo(d) > 0)) {
                            result = d;
                        }
                        break;
                    }
                }
            }
        }

        return result;
    }
}